src/app/login/login.component.ts

```diff
@@ -7,9 +9,42 @@ import { Component, OnInit } from '@angular/core';
 })
 export class LoginComponent implements OnInit {
 
-  constructor() { }
+  loginForm: FormGroup; 
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private router: Router
+    ) { 
+  }
 
   ngOnInit(): void {
+    this.loginForm = this.formBuilder.group({
+      username: ['', Validators.required],
+      password: ['', Validators.required]
+    })
   }
 
+  submit(){
+    if(this.loginForm.valid){
+      console.log("Valid");
+
+      const username = this.loginForm.get('username').value;
+      console.log(username);
+
+      if(username === "teacher"){
+        console.log("Yes: Teacher");
+        this.router.navigateByUrl('/teacher');
+        
+      }
+      else if (username === "student"){
+        console.log("Yes: Student");
+        this.router.navigateByUrl('/students')
+        
+      }
+    }
+    else {
+      console.log("Unvalid");
```

Comment:
Should say "invalid". Unvalid is not a word

Reply:
Josh's English.